### PR TITLE
pytorch 2.1.2

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@jaimergp: could you repush an empty commit, please?

**Comment by jjerphan** [comment]
Thank you, @jaimergp.

It seems that one workflow is marked as failed, but it is not possible to access its logs. 

Should we rerun all of them?

**Comment by jaimergp** [comment]
I've rerun failures once. I think we are close to resource exhaustion and some workers die. Let's give it a 2nd try.

**Comment by jjerphan** [comment]
I think there is a warning which is duplicated multiple times and which spams the logs showing the job as failing even-though the job itself might not fail.

**Comment by isuruf** [comment]
Yeah, @hmaarrfk has been building that job manually.

File: .ci_support/linux_64_blas_implgenericc_compiler_version10cuda_compilernvcccuda_compiler_version11.2cxx_compiler_version10.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ python:
 - 3.8.* *_cpython
 - 3.9.* *_cpython
 pytorch:
-- '2.0'
+- '2.1'
```
Comment (jjerphan) [inline .ci_support/linux_64_blas_implgenericc_compiler_version10cuda_compilernvcccuda_compiler_version11.2cxx_compiler_version10.yaml:56]: Might the previous builds for `2.1.0` be compromised?

Comment (hmaarrfk) [reply .ci_support/linux_64_blas_implgenericc_compiler_version10cuda_compilernvcccuda_compiler_version11.2cxx_compiler_version10.yaml:56]: I think this pin is somewhat ignored. This funny thing is happening in rendering this repo that is odd with the PyTorch pinning

Comment (jjerphan) [reply .ci_support/linux_64_blas_implgenericc_compiler_version10cuda_compilernvcccuda_compiler_version11.2cxx_compiler_version10.yaml:56]: I find the behaviour of conda build misleading.

**Comment by jjerphan** [comment]
Thank you, @hmaarrfk.

**Pull request merged**